GOT-StudentVersion/GOT-StudentVersion/Control/ViewController.swift

```diff
@@ -0,0 +1,174 @@
+    @IBOutlet weak var gotSearchBar: UISearchBar!
+    
+    let gotSeasons = GOTEpisode.allSeasons
+    //will return an array, when calling data in function, use indexpath to get the element needed
+    var episodes = GOTEpisode.allEpisodes {
+        didSet {
+            gotTableView.reloadData()
+        }
+    }
+    
+    
+  
+    var gotSearchResult: [[GOTEpisode]] {
+        get {
+            guard let searchString = searchString else {
+                return gotSeasons
+            }
+            guard searchString != "" else {
+                return gotSeasons
+            }
+            if let scopeTitles = gotSearchBar.scopeButtonTitles {
+                let currentScopeIndex = gotSearchBar.selectedScopeButtonIndex
+                
+                switch scopeTitles[currentScopeIndex] {
+                case "Ep. Name":
+                    return [gotSeasons.joined().filter{$0.name.lowercased().contains(searchString.lowercased())}]
+                case "Summary":
+                    return [gotSeasons.joined().filter{$0.summary.lowercased().contains(searchString.lowercased())}]
+                default:
+                    return gotSeasons                }
+            }
+            return gotSeasons
+        }
+    }
+    
+    var searchString: String? = nil {
+        didSet {
+            print(searchString as Any)
+            //this function makes the tableView
+            gotTableView.reloadData()
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupProtocols()
+    }
+    
+    
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        //if/else
+        guard let destination = segue.destination as? DetailViewController,
+            let cellSelected = gotTableView.indexPathForSelectedRow else {return}
+        let episodeSelected = gotSeasons[cellSelected.section][cellSelected.row]
+        
+        
+        if segue.identifier == "oddEpisode"  {
+            destination.gotData = episodeSelected
+        } else if segue.identifier == "evenEpisode" {
+            destination.gotData = episodeSelected
+        }
+    }
+    
+    
+    
+    
+    private func setupProtocols() {
+        gotTableView.delegate = self
+        gotTableView.dataSource = self
+        gotSearchBar.delegate = self
+    }
+}
+
+extension ViewController: UITableViewDataSource, UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return gotSearchResult.count == 0 ?  1 : gotSearchResult.count
+//        return gotSeasons[section].count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let gotSeason = gotSearchResult[indexPath.section][indexPath.row]
+        let customView = UIView()
+        customView.backgroundColor = UIColor(red: 59/255, green: 90/255, blue: 99/255, alpha: 0.5)
```

Comment:
Avoid adding view properties locally, next time create a file where you have all the UI elements and call it here